classes/data-stores/ActionScheduler_DBStore.php

```diff
@@ -993,36 +994,90 @@ protected function claim_actions( $claim_id, $limit, ?DateTime $before_date = nu
 			$where  .= " AND group_id {$group_operator} ( $id_list )";
 		}
 
-		/**
-		 * Sets the order-by clause used in the action claim query.
-		 *
-		 * @since 3.4.0
-		 * @since 3.8.3 Made $claim_id and $hooks available.
-		 *
-		 * @param string $order_by_sql
-		 * @param string $claim_id Claim Id.
-		 * @param array  $hooks Hooks to filter for.
-		 */
-		$order    = apply_filters( 'action_scheduler_claim_actions_order_by', 'ORDER BY priority ASC, attempts ASC, scheduled_date_gmt ASC, action_id ASC', $claim_id, $hooks );
-		$params[] = $limit;
-
-		$sql           = $wpdb->prepare( "{$update} {$where} {$order} LIMIT %d", $params ); // phpcs:ignore WordPress.DB.PreparedSQL.InterpolatedNotPrepared, WordPress.DB.PreparedSQLPlaceholders
-		$rows_affected = $wpdb->query( $sql ); // phpcs:ignore WordPress.DB.PreparedSQL.NotPrepared, WordPress.DB.DirectDatabaseQuery.DirectQuery, WordPress.DB.DirectDatabaseQuery.NoCaching
-		if ( false === $rows_affected ) {
-			$error = empty( $wpdb->last_error )
-				? _x( 'unknown', 'database error', 'action-scheduler' )
-				: $wpdb->last_error;
+		$use_priorities = true;
```

Comment:
I added this variable and flow clause below specifically for simplifying comparing the performance of the two operations.